### ISSUE TITLE
Accept lists of ports

### ISSUE THREAD

**Issue by sholladay** [baalexander/node-portscanner#26]
The APIs for this module would be much more useful if they could also consume lists of ports that we deem acceptable, in addition to the current `startPort` and `endPort` abilities. This will help prevent accidentally hogging specific ports, and therefor clashing with other services used in our projects, without forcing us to restrict the scanning range to something above or below those points. For example, I should be able to instruct `portscanner` to choose one out of _exactly_ _n_ number of specific ports.

``` javascript
portscanner.findAPortNotInUse([4443, 4444, 4447], '127.0.0.1', function(error, port) {
    console.log('Available port: ' + port);
});
```

And that should never give me a port I did not list explicitly.

**Comment by sholladay** [comment]
FYI, I have posted a bounty for this issue on Bountysource.
https://www.bountysource.com/issues/7446655-accept-lists-of-ports

**Comment by darthneko** [comment]
Modified existing methods to work with a list of ports instead of a starting port and ending port. https://github.com/baalexander/node-portscanner/pull/28

**Comment by darthneko** [comment]
Revised suggested changes. #28

**Comment by darthneko** [comment]
Please, review this pull request and either state why functionality isn't right or approve closing of the issue. #28

**Comment by sholladay** [comment]
Edited to clarify that my desired behavior is to add the ability to consume lists, without replacing or destroying the current `startPort` and `endPort` abilities. They should not be mutually exclusive. I see no reason this has to be a breaking change.

**Comment by darthneko** [comment]
It's the exact functionality of this branch.

**Comment by laggingreflex** [comment]
This has been added. Released a new minor version 1.1.0, please check it out and let me know.

**Comment by laggingreflex** [comment]
It turns out v1.1.0 broke portscanner in some cases and the changes that broke it were actually the changes required to fix this exact issue. For that reason I've published the new changes as v2.0.0. I'm sorry this feature couldn't make it a minor version. [I've addressed the issue here](https://github.com/baalexander/node-portscanner/issues/51).
